hacspec/examples/aes-gcm/aes.rs

```diff
@@ -90,43 +104,59 @@ fn mix_columns(state: Block) -> Block {
     mix_column(3, state)
 }
 
-fn add_round_key(state: Block, key: Key) -> Block {
+fn add_round_key(state: Block, key: RoundKey) -> Block {
     let mut out = state;
     for i in 0..BLOCKSIZE {
         out[i] ^= key[i];
     }
     out
 }
 
-fn aes_enc(state: Block, round_key: Key) -> Block {
+fn aes_enc(state: Block, round_key: RoundKey) -> Block {
     let state = sub_bytes(state);
     let state = shift_rows(state);
     let state = mix_columns(state);
     add_round_key(state, round_key)
 }
 
-fn aes_enc_last(state: Block, round_key: Key) -> Block {
+fn aes_enc_last(state: Block, round_key: RoundKey) -> Block {
     let state = sub_bytes(state);
     let state = shift_rows(state);
     add_round_key(state, round_key)
 }
 
-fn rounds(state: Block, key: Bytes144) -> Block {
+fn rounds_aes128(state: Block, key: Bytes144) -> Block {
+    let mut out = state;
+    for (_, key_block) in key.chunks(BLOCKSIZE) {
+        out = aes_enc(out, RoundKey::from(key_block));
+    }
+    out
+}
+fn rounds_aes256(state: Block, key: Bytes208) -> Block {
     let mut out = state;
     for (_, key_block) in key.chunks(BLOCKSIZE) {
-        out = aes_enc(out, Key::from(key_block));
+        out = aes_enc(out, RoundKey::from(key_block));
     }
     out
 }
 
-fn block_cipher(input: Block, key: Bytes176) -> Block {
-    let k0 = Key::from_sub(key, 0..16);
-    let k = Bytes144::from_sub(key, 16..10 * 16);
-    let kn = Key::from_sub(key, 10 * 16..11 * 16);
+fn block_cipher_aes128(input: Block, key: Bytes176, nr:usize) -> Block {
```

Comment:
Why do we need `nr` now? As long as we have separate functions for 128 and 256 there should be no need for `nr` or `nk` parameters I think.

Reply:
Right, 
it was part of my first try ( where I wanted both versions to run the same functions) and I thought it would be nice for orientation but I can delete it.

Reply:
Both aes use the same key_expansion_word() where nr and nk are needed